libs/components/core/testing/src/shared/component-harness.ts

```diff
@@ -1,10 +1,25 @@
-import { ComponentHarness } from '@angular/cdk/testing';
+import {
+  ComponentHarness,
+  ComponentHarnessConstructor,
+  HarnessPredicate,
+} from '@angular/cdk/testing';
+
+import { SkyHarnessFilters } from './harness-filters';
 
-/**
- * @experimental
- */
 export abstract class SkyComponentHarness extends ComponentHarness {
-  protected async getSkyId(): Promise<string | null> {
+  protected static getDataSkyIdPredicate<T extends SkyComponentHarness>(
+    this: ComponentHarnessConstructor<T>,
+    filters: SkyHarnessFilters
+  ) {
+    return new HarnessPredicate(this, filters).addOption(
+      'dataSkyId',
+      filters.dataSkyId,
+      (harness, text) =>
+        HarnessPredicate.stringMatches(harness.#getSkyId(), text)
+    );
+  }
+
+  async #getSkyId(): Promise<string | null> {
```

Comment:
Is this meant to be public now? If so, you should specify the `public` access modifier.

Reply:
It's meant to be "private private". Is that a problem?

Reply:
🤦 It's fine. Still too early in the morning for me I guess.